blurring_as_a_service/inference_pipeline/components/detect_and_blur_sensitive_data.py

```diff
@@ -91,46 +96,64 @@ def detect_and_blur_sensitive_data(
+        os.makedirs(locked_folder, exist_ok=True)
+
+        # Check if the path points to a file (not a directory) and if the file exists
+        if os.path.isfile(file_path) and os.path.exists(file_path):
+            logger.info(f"Creating inference step: {file_path}")
 
             files_to_blur_full_path = os.path.join(
                 yolo_yaml_path, batch_file_txt
             )  # use outputs folder as Azure expects outputs there
-            with open(file_path, "r") as src:
-                with open(files_to_blur_full_path, "w") as dest:
-                    for line in src:
-                        dest.write(f"{input_structured_folder}/{line}")
-                        print(f"{input_structured_folder}/{line}")
-
-            data = dict(
-                train=f"{files_to_blur_full_path}",
-                val=f"{files_to_blur_full_path}",
-                test=f"{files_to_blur_full_path}",
-                nc=2,
-                names=["person", "license_plate"],
-            )
-
-            # TODO create postgresql string and send to val.py
-
-            with open(f"{yolo_yaml_path}/pano.yaml", "w") as outfile:
-                yaml.dump(data, outfile, default_flow_style=False)
-            cuda_device = torch.cuda.current_device()
-            model_parameters = json.loads(model_parameters_json)
-            database_parameters = json.loads(database_parameters_json)
-            val.run(
-                weights=model,
-                data=f"{yolo_yaml_path}/pano.yaml",
-                project=results_path,
-                device=cuda_device,
-                name="",
-                customer_name=customer_name,
-                start_time=get_current_time(),
-                run_id=generate_unique_string(10),
-                **model_parameters,
-                **database_parameters,
-            )
-            delete_file(file_path)
-            delete_file(files_to_blur_full_path)
-            delete_file(f"{yolo_yaml_path}/pano.yaml")
+
+            read_success = False
+            try:
+                shutil.move(file_path, locked_path)
+
+                with open(locked_path, "r") as src:
+                    with open(files_to_blur_full_path, "w") as dest:
+                        for line in src:
+                            dest.write(f"{input_structured_folder}/{line}")
+                read_success = True
+            except FileNotFoundError as e:
+                logger.info(f"File {locked_path} not found: {e}")
+            except Exception as e:
+                logger.error(f"Error occurred while reading {locked_path}: {e}")
+
+            if read_success:
+                data = dict(
+                    train=f"{files_to_blur_full_path}",
+                    val=f"{files_to_blur_full_path}",
+                    test=f"{files_to_blur_full_path}",
+                    nc=2,
+                    names=["person", "license_plate"],
+                )
+
+                # Remove the extension
```

Comment:
The idea works, but instead of using this renaming technique, shall we use a package like the following?
https://py-filelock.readthedocs.io/en/latest/